sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -374,6 +374,29 @@ static std::string getUint32PropAsOptStr(const RTDeviceBinaryImage &Img,
   return temp;
 }
 
+static void appendCompileOptionsForRegAllocMode(std::string &CompileOpts,
+                                                const RTDeviceBinaryImage &Img,
+                                                bool IsEsimdImage) {
+  pi_device_binary_property Prop = Img.getProperty("RegisterAllocMode");
+  if (!Prop)
+    return;
+  uint32_t PropVal = DeviceBinaryProperty(Prop).asUint32();
+  // 2 means Large GRF.
+  if (PropVal == 2) {
```

Comment:
Could we do something like

```suggestion
  if (PropVal == static_cast<uint32_t>(ext::oneapi::experimental::register_alloc_mode_enum::large)) {
```

instead? Granted it is more verbose, I think it explains the intention as nicely as the comment in the line above.

This uses the naming change in my previous comment.

Reply:
done in latest commit thx